keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,727 @@
+
+##### e2e tests
+
+TBD
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind feature gates (`DRAResourceClaimDeviceStatus`).
+  Feature Gates are disabled by default.
+- Documentation provided.
+- Initial unit, integration and e2e tests completed and enabled.
+- Authorization implemented to allow only the user on the same node as the 
+  allocated `ResourceClaim` to write the status of the devices.
+
+#### Beta
+
+- Feature Gates are enabled by default.
+- No major outstanding bugs.
+- Feedback collected from the community (developers and users) with adjustments
+  provided, implemented and tested.
+
+#### GA
+
+- 2 examples of real-world usage.
+- Allowing time for feedback from developers and users.
+
+### Upgrade / Downgrade Strategy
+
+This feature only exposes a new field in the `ResourceClaim.Status`, the field
+will either be present or not.
+
+DRA implementation requires DRA interfaces change. DRA is in alpha and in
+active development. The feature will follow the DRA upgrade/downgrade strategy.
+
+### Version Skew Strategy
+
+This feature affects only the kube-apiserver, so there is no issue with version
+skew with other Kubernetes components.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DRAResourceClaimDeviceStatus
+  - Components depending on the feature gate: kube-apiserver
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning of
+    a node?
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, with no side effects except for missing the new field in
+`ResourceClaim.Status`. Re-enabling this feature will not guarantee to keep
+the values written before the feature has been disabled.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+###### Are there any tests for feature enablement/disablement?
+
+Enablement/disablement of this feature is tested as part of the integration
+tests.
+
+### Rollout, Upgrade and Rollback Planning
+
+No
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+No
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Check the `ResourceClaim.Status.Devices`.
```

Comment:
Typically we want a metric for this. A count of ResourceClaims with this field set? Can you get that from kube state metrics?

Reply:
resourceclaim_status_devices_update_attempts_total and resourceclaim_status_devices_update_failures_total have been added. I am not sure about kube state metrics, I could take a look at it, but this will be directly implemented in kube-apiserver right?